components/camel-rest-swagger/src/main/java/org/apache/camel/component/rest/swagger/RestSwaggerEndpoint.java

```diff
@@ -165,7 +177,18 @@ public Consumer createConsumer(final Processor processor) throws Exception {
     @Override
     public Producer createProducer() throws Exception {
         final CamelContext camelContext = getCamelContext();
-        final Swagger swagger = loadSpecificationFrom(camelContext, specificationUri);
+        SSLContextParameters sslContextParameters = getSslContextParameters();
+        if (sslContextParameters == null) {
+            sslContextParameters = component().getSslContextParameters();
+        }
+        if (sslContextParameters == null) {
+            sslContextParameters = camelContext.getSSLContextParameters();
+        }
+        if (sslContextParameters == null) {
+            sslContextParameters = component().retrieveGlobalSslContextParameters();
```

Comment:
Do we check if `retrieveGlobalSslContextParameters` doesn't return `null` somewhere?

Reply:
sslContextParameters could be null. (and it will be in majority of calls). Null means that the security is not defined http-component won't be used - https://github.com/apache/camel/pull/3374/files#diff-198a2a776971f53def53c61f07af2721R633